src/teleop_twist_joy.cpp

```diff
@@ -68,63 +68,73 @@ struct TeleopTwistJoy::Impl
-    node->get_parameter("scale_linear", pimpl_->scale_linear_map["x"]);
-    pimpl_->scale_linear_turbo_map["x"] = 1.0;
-    node->get_parameter("scale_linear_turbo", pimpl_->scale_linear_turbo_map["x"]);
+  // TODO(clalancette): node->get_parameter(s) doesn't support getting a map of
+  // values yet.  Revisit this once it does.
+  int_parameters.push_back({"enable_button", 5, pimpl_->enable_button});
+  int_parameters.push_back({"enable_turbo_button", -1, pimpl_->enable_turbo_button});
+  int_parameters.push_back({"axis_linear_x", 5, pimpl_->axis_linear_map["x"]});
+  int_parameters.push_back({"axis_angular_yaw", 2, pimpl_->axis_angular_map["yaw"]});
+
+  double_parameters.push_back({"scale_linear_x", 0.5, pimpl_->scale_linear_map["x"]});
+  double_parameters.push_back({"scale_linear_turbo_x", 1.0, pimpl_->scale_linear_turbo_map["x"]});
+  double_parameters.push_back({"scale_angular_yaw", 0.5, pimpl_->scale_angular_map["yaw"]});
+  double_parameters.push_back({"scale_angular_turbo_yaw", 0.5, pimpl_->scale_angular_turbo_map["yaw"]});
+
+  for(const auto& p : int_parameters) {
+    this->set_parameter_if_not_set(p.name, p.default_value);
+    this->get_parameter(p.name, p.variable);
   }
 
-  // TODO(clalancette): node->get_parameter(s) doesn't seem to
-  // support getting a map of values yet.  Revisit this once it does.
-  // if (nh_param->getParam("axis_angular", pimpl_->axis_angular_map))
-  // {
-  //   nh_param->getParam("axis_angular", pimpl_->axis_angular_map);
-  //   nh_param->getParam("scale_angular", pimpl_->scale_angular_map);
-  //   nh_param->getParam("scale_angular_turbo", pimpl_->scale_angular_turbo_map);
-  // }
-  // else
-  {
-    pimpl_->axis_angular_map["yaw"] = 2;
-    node->get_parameter("axis_angular", pimpl_->axis_angular_map["yaw"]);
-    pimpl_->scale_angular_map["yaw"] = 0.5;
-    node->get_parameter("scale_angular", pimpl_->scale_angular_map["yaw"]);
-    pimpl_->scale_angular_turbo_map["yaw"] = pimpl_->scale_angular_map["yaw"];
-    node->get_parameter("scale_angular_turbo", pimpl_->scale_angular_turbo_map["yaw"]);
+  for(const auto& p : double_parameters) {
+    this->set_parameter_if_not_set(p.name, p.default_value);
+    this->get_parameter(p.name, p.variable);
   }
 
+  auto param_change_callback =
+    [this](std::vector<rclcpp::Parameter> parameters) -> rcl_interfaces::msg::SetParametersResult
+    {
+      rcl_interfaces::msg::SetParametersResult result;
+      result.successful = true;
+      result.reason = "";
+      for (const auto& param : parameters) {
+        try {
+          std::string name = param.get_name();
+          auto int_it = std::find_if(int_parameters.begin(), int_parameters.end(),
+                                     [&name](const JoyParameter<int>& j) -> bool {
+                                       return name == j.name;
+                                     });
+          if (int_it != int_parameters.end()) {
+            int_it->variable = param.as_int();
```

Comment:
Should this print the new value like it's done when the node starts? (same below)

Reply:
That's a good idea, I'll move the printing to a method on the `Impl` class.